### ISSUE TITLE
Feature request: Add Icon to Autocomplete

### ISSUE THREAD

**Issue by timschoch** [SchweizerischeBundesbahnen/sbb-angular#121]
**Is your feature request related to a problem? Please describe.**
We have multiple autocompletes next to each other in our filter panel and all look similar. They are grouped to help with understanding and orientation, but it's still hard distinguish them as the list grows. Adding an icon would let the user know at a glance what they are searching (waggon icon, station, people, ...)

**Describe the solution you'd like**
Set an optional icon for the autocomplete, styled like the search box

**Describe alternatives you've considered**
Using the label helps a bit, but still forces the user to read

**Additional context**
![grafik](https://user-images.githubusercontent.com/10527318/62216667-7842cf00-b3a9-11e9-859d-1db0e149d7ae.png)

![grafik](https://user-images.githubusercontent.com/10527318/62216650-71b45780-b3a9-11e9-87c6-841edb89aabb.png)

**Comment by kyubisation** [comment]
If I understand correctly, you would like to use a random (appropriate) icon to appear in the autocomplete input.
I assume text behind the icon should be hidden?

**Comment by timschoch** [comment]
Exactly. It would work like the search, just without the red background:
https://angular.app.sbb.ch/latest/content/search

Maybe even inappropriate icons too... harrharr...

**Comment by timschoch** [comment]
Can I bump this up again and challenge the low priority?
We're building relatively complex order forms where our users have to add several different type of information. This is a snippet:
![grafik](https://user-images.githubusercontent.com/10527318/68133079-0cc84a80-ff20-11e9-997f-b3d00d42d050.png)
To help the user comprehend what is expected from them, we want to add a distinct icon per autocomplete data type. (eg. to add contacts we'll use a user+magnifying glass).

**Comment by kyubisation** [comment]
The solution to this is relatively simple.
Implement a CSS class (e.g. `sbb-icon-hint`, which can be placed on the input. This class sets the padding-right to an appropriate size and moves a following icon to the left.
```html
<input sbbInput [sbbAutocomplete]="autocomplete" class="sbb-icon-hint"><sbb-icon-random-icon></sbb-icon-random-icon>
```

However, I'm currently busy with refactoring the header and afterwards I will be busy with bugs.
The low priority is due to the fact, that this can be implemented by consumers themselves.
If you can provide a contribution to the business typography, I will certainly have a look at it.

@Frenggi This seems to be a reasonable addition to the design system. Do you agree?

**Comment by timschoch** [comment]
@kyubisation Since we're using this in several places (filter panels too) I'll have a chat with our devs and see if we can provide a PR - provided @frenggi decides this should be in the design system. 

What do you mean by contributing to the business typography? (sorry, if we've talked about this. I'm tired and can't remember... ;) )

**Comment by kyubisation** [comment]
Don't worry about it. I'm sorry for the limited support at the moment. I hope to improve this next year.

In regards to "business typography"; This was a technical categorization. But that might have been wrong. Sorry about that.
It should be implemented in autocomplete.component.scss, wrapped in a `@include businessOnly() { ... }`.

**Comment by timschoch** [comment]
No worries about the support, I know you're giving it your best and - judging from the release times - doing it in your spare time. Let's wait for a UX feedback, and we'll see how we can take it from there.

**Comment by kyubisation** [comment]
@Frenggi fyi
